backend/src/lib/telemetry/instrumentation.ts

```diff
@@ -0,0 +1,46 @@
+import opentelemetry from "@opentelemetry/api";
+import { getNodeAutoInstrumentations } from "@opentelemetry/auto-instrumentations-node";
+import { OTLPMetricExporter } from "@opentelemetry/exporter-metrics-otlp-proto";
+import { PrometheusExporter } from "@opentelemetry/exporter-prometheus";
+import { registerInstrumentations } from "@opentelemetry/instrumentation";
+import { Resource } from "@opentelemetry/resources";
+import { AggregationTemporality, MeterProvider, PeriodicExportingMetricReader } from "@opentelemetry/sdk-metrics";
+import { SEMRESATTRS_SERVICE_NAME, SEMRESATTRS_SERVICE_VERSION } from "@opentelemetry/semantic-conventions";
+
+export const initTelemetry = async ({ otlpURL }: { otlpURL?: string }) => {
+  const resource = Resource.default().merge(
+    new Resource({
+      [SEMRESATTRS_SERVICE_NAME]: "infisical-server",
+      [SEMRESATTRS_SERVICE_VERSION]: "0.1.0"
+    })
+  );
+
+  const metricReaders = [];
+  if (otlpURL) {
+    const otlpExporter = new OTLPMetricExporter({
+      url: `${otlpURL}/v1/metrics`,
```

Comment:
I see that we only added the endpoint but i assume most of these end points are secured by bear tokens or other header options? Maybe we should add at least a header option such as 

```
export OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer an_apm_secret_token"
```

Reply:
Also question, is it always `/v1/metrics`?

Reply:
yup! `/v1/metrics` for metrics. 

it's defined in the OTEL specs